src/openrct2/network/network.cpp

```diff
@@ -2134,7 +2134,11 @@ sint32 network_begin_client(const char *host, sint32 port)
 
 sint32 network_begin_server(sint32 port)
 {
-	return gNetwork.BeginServer(port);
+	char *address = nullptr;
+	if (strlen(gConfigNetwork.listen_address) > 0)
+		address = gConfigNetwork.listen_address;
+
+	return gNetwork.BeginServer(port, address);
```

Comment:
I think the empty string check should go here:
https://github.com/OpenRCT2/OpenRCT2/blob/c4e89278c111d3c3a75f199d58496bc5d388795f/src/openrct2/network/TcpSocket.cpp#L457